src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -743,12 +743,18 @@ trait RuntimeCosting extends SigmaLibrary with DataCosting {
             withDefaultSize(se.fields.length, xsC.cost + costOf(node))
         }
 
-      case ByIndex(xs, i, None) =>
+      case ByIndex(xs, i, default) =>
         val xsC = evalNode(ctx, env, xs).asRep[CostedCol[Any]]
         val iC = evalNode(ctx, env, i).asRep[Costed[Int]]
         val iV = iC.value
         val size = xsC.sizes(iV)
-        CostedPrimRep(xsC.value(iV), xsC.cost + iC.cost + costOf(node), size)
+        default match {
```

Comment:
@ergomorphic Could you please check out my first take on costing?

Reply:
https://github.com/ScorexFoundation/sigmastate-interpreter/commit/6b06988eaba7b11bd42ce8d469f328b986bd5754